synapse/api/room_versions.py

```diff
@@ -69,6 +69,8 @@ class RoomVersion:
     limit_notifications_power_levels = attr.ib(type=bool)
     # MSC2174/MSC2176: Apply updated redaction rules algorithm.
     msc2176_redaction_rules = attr.ib(type=bool)
+    # MSC2962: Support the 'restricted' join_rule.
+    msc2962_join_rules = attr.ib(type=bool)
```

Comment:
This is currently MSC2962, but I think we're going to split it out. Maybe I should put up a PR to the draft to reserve a number?

Reply:
maybe do the same as https://github.com/matrix-org/matrix-doc/blob/kegan/spaces-summary/proposals/2946-spaces-summary.md

Reply:
See matrix-org/matrix-doc#3083